src/activation.cpp

```diff
@@ -231,46 +214,38 @@ void mixture_relu_mean_var_mp(std::vector<float> &mu_z,
 }
 
 void mixture_sigmoid_mean_var(std::vector<float> &mu_z,
-                              std::vector<float> &var_z, float omega_tol,
+                              std::vector<float> &var_z,
                               int start_chunk, int end_chunk,
                               std::vector<float> &mu_a, std::vector<float> &jcb,
                               std::vector<float> &var_a)
 /*
  */
 {
-    float alpha_lower, alpha_upper, omega, beta, kappa, mu_z_til, var_z_til,
-        cdf_lower, cdf_upper, pdf_lower, pdf_upper;
+    float std_z, alpha_l, alpha_u, pdf_l, pdf_u, cdf_l, cdf_u;
     for (int i = start_chunk; i < end_chunk; i++) {
-        alpha_lower = (-1.0f - mu_z[i]) / powf(var_z[i], 0.5);
-        alpha_upper = (1.0f - mu_z[i]) / powf(var_z[i], 0.5);
-        cdf_lower = normcdf_cpu(alpha_lower);
-        cdf_upper = normcdf_cpu(alpha_upper);
-        pdf_lower = normpdf_cpu(alpha_lower, 0.0f, 1.0f);
-        pdf_upper = normpdf_cpu(alpha_upper, 0.0f, 1.0f);
-
-        // Truncated distribution's parameters
-        omega = std::max(cdf_upper - cdf_lower, omega_tol);
-        beta = (pdf_upper - pdf_lower) / omega;
-        kappa = 1 -
-                ((pdf_upper * alpha_upper - pdf_lower * alpha_lower) / omega) -
-                powf(beta, 2);
-
-        // Gaussian mixture's paramters
-        mu_z_til = mu_z[i] - beta * pow(var_z[i], 0.5);
-        var_z_til = kappa * var_z[i];
-
-        // Activation distribution
-        mu_a[i] =
-            (omega * mu_z_til - cdf_lower + (1 - cdf_upper)) / 2.0f + 0.5f;
-        var_a[i] = (omega * var_z_til + omega * powf(mu_z_til - mu_a[i], 2) +
-                    cdf_lower * powf(1 + mu_a[i], 2) +
-                    (1 - cdf_upper) * powf(1 - mu_a[i], 2)) /
-                   4.0f;
-        jcb[i] = omega * 0.5;
+        // cdf and pdf for truncated normal distribution
+        std_z = powf(var_z[i], 0.5);
+        alpha_l = (1.0f + mu_z[i]) / std_z;  // Lower truncation
+        alpha_u = (1.0f - mu_z[i]) / std_z;  // Upper truncation
+        cdf_l = normcdf_cpu(alpha_l);
+        cdf_u = normcdf_cpu(alpha_u);
+        pdf_l = normpdf_cpu(alpha_l, 0.0f, 1.0f);
+        pdf_u = normpdf_cpu(alpha_u, 0.0f, 1.0f);
+
+        // Moments calculations (L. Alric, 2024)
+        mu_a[i] = (mu_z[i] + 1) * cdf_l + (mu_z[i] - 1) * cdf_u +
+                   std_z * (pdf_l - pdf_u) - mu_z[i];
+        var_a[i] = (cdf_l * (var_z[i] - powf(mu_z[i], 2) - 2 * mu_z[i] - 1) +
```

Comment:
@jamesgoulet Does it improve the results compared to the previous version? There is way more computations than the previous version.

Reply:
@jamesgoulet could you please remove the omega_tol in `include/activation.h` and `include/activation_cuda.cuh`? Could you please provide the relationship equation between A and Z in pdf file?

Reply:
> @jamesgoulet could you please remove the omega_tol in `include/activation.h` and `include/activation_cuda.cuh`? 

@lhnguyen102 I do not get your comment regarding the removal of `omega_tol` in `include/activation.h` and `include/activation_cuda.cuh`, as I doubled check and they are already removed from both files. 
<img width="1090" alt="Screenshot 2024-04-07 at 16 37 06" src="https://github.com/lhnguyen102/cuTAGI/assets/26071993/70ff0e2e-ff1d-4539-9582-2eb53bb88321">
<img width="1078" alt="Screenshot 2024-04-07 at 16 37 26" src="https://github.com/lhnguyen102/cuTAGI/assets/26071993/ef09b7f8-c960-4afe-a852-52d2dcbd106a">
Maybe I did not correctly understood your question?

Reply:
> @jamesgoulet  Could you please provide the relationship equation between A and Z in pdf file?

@lhnguyen102 I have modified and re-upload the PDF with the relationship equation between A and Z:
<img width="237" alt="Screenshot 2024-04-07 at 16 52 14" src="https://github.com/lhnguyen102/cuTAGI/assets/26071993/ff03a2ca-109c-4a9c-ac09-64023e7a02f5">

<img width="239" alt="Screenshot 2024-04-07 at 16 52 19" src="https://github.com/lhnguyen102/cuTAGI/assets/26071993/a06d4c4a-1c45-456c-b882-1163475605d0">

Reply:
> @jamesgoulet Does it improve the results compared to the previous version? There is way more computations than the previous version.

In the checks I have done, the performance was on par with the previous one (except for mSigmoid, in which case there was a typo in the previous version). The issue with `omega_tol` was a lack of robustness as pointed on by @zhanwen-xin earlier experiments who needed to play with the value to get a good performance. With the exact formulation, there is nothing to tune and we are not constantly in the waiting for a case where the formulation could fail. 

Regarding the increase in computation, it is indeed the case for mTanh and mSigmoid, but not really for mRelu. Moreover, I think that these analytical activations should be reserved for the uncertainty propagation case, i.e. PORL, as they tend to underperform the simpler and cheaper 1st order Taylor approximation, i.e., local linearization. Since UC propagation requires full covariances for hidden layers, such additional compute in the activations is not the bottleneck.

Let me know to what extent you think the additional compute will be an issue.

Reply:
Thank you for detailed reply. My bad, I thought you did not remove `omega_tol` in those two files but you did.  It is all good on my side.